Mesh_3/include/CGAL/Polyhedral_mesh_domain_with_features_3.h

```diff
@@ -234,8 +243,47 @@ detect_features(FT angle_in_degree, Polyhedron& p)
   this->add_features(
     boost::make_transform_iterator(polylines.begin(),extractor),
     boost::make_transform_iterator(polylines.end(),extractor));
+
+  detect_borders();
+}
+
+
+template < typename GT_, typename P_, typename TA_,
+           typename Tag_, typename E_tag_>
+void
+Polyhedral_mesh_domain_with_features_3<GT_,P_,TA_,Tag_,E_tag_>::
+detect_borders()
+{
+  if (borders_detected_)
+    return;//border detection has already been done
+
+  typedef std::vector<Point_3> Polyline;
+  typedef std::vector<Polyline>Polylines;
+
+  Polylines polylines;
+  Polyline empty;
+  typedef boost::graph_traits<Polyhedron>::halfedge_descriptor halfedge_descriptor;
+  std::set<halfedge_descriptor> visited;
+  BOOST_FOREACH(halfedge_descriptor h, halfedges(polyhedron_)){
```

Comment:
this code could be partly replaced by the [`PMP::border_halfedges()`](http://doc.cgal.org/latest/Polygon_mesh_processing/group__PkgPolygonMeshProcessing.html#gaf5f5eb322811f24573aa9158e6df8070) 
@afabri @lrineau do you think we should use this function here?